README.md

```diff
@@ -2,12 +2,103 @@
+```
+This will create the Node Components and Ethernet Interfaces. If `--fake-discovery` is added to the above and a `bmc_ipaddr` is defined for each node in the config, Redfish Endpoints and ComponentEndpoints will also be created. This will "simulate" hardware discovery. 
+
+Nodes can be added manually with
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --add-node --xname <xname> --nid <nid>
+```
+and deleted with
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --delete-node --xname <xname>
+```
+You can also get the components in SMD with 
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --get-components --format {none|json|yaml}
+```
+or specify an `xname` or `nid`
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --get-components --xname <xname>
+```
+
+Ethernet Interfaces can also be added manually with:
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --add-interface --name <name> --xname <xname> --mac <mac> --ipaddr <ipaddr>
+```
+and deleted with
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --delete-interface --xname <xname>
+```
+Get all the EthernetInterfaces:
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --get-interfaces --format {none|json|yaml}
+```
+or specify an xname
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --get-interfaces--xname <xname>
+```
+
+You can dump SMD into yaml format that can then be read back into SMD. Useful for saving as a backup or saving updates done with the cmdline.
+```bash
+./ochami-cli --smd-url http://smd:27779/hsm/v2 smd --dump > nodes.yaml
+```
+
+Here is an example nodes.yaml file:
+```yaml
+nodes:
+  - name: cg01
+    xname: x1000c1s7b0n0
+    mac: b4:...
+    ipaddr: 172.16.0.1
+    bmc_ipaddr: 172.16.100.1
+  - name: cg02
+    xname: x1000c1s7b1n0
+    mac: b4:...
+    ipaddr: 172.16.0.2
+    bmc_ipaddr: 172.16.100.1
+```
+### BSS
+
+BSS can be populated with a config file:
+```bash
+./ochami-cli --bss-url http://bss:27778/boot/v1 bss --config bss.yml --add-bootparams
+```
+or updated
+```bash
+./ochami-cli --bss-url http://bss:27778/boot/v1 bss --config bss.yml --update-bootparams
+```
 
+Individual components can be updates as well, you can choose one or more of the following after specifying `--update-bootparams`; `--kernel`, `--initrd`, `--image`, `--params`.
+Updating the `--image` and `--params` at the same time will result in the `--params` overwriting what is specified in `--image`
 ```bash
-./ochami-cli smd --config config.yml --url http://smd:27779/hsm/v2
+./ochami-cli --bss-url http://bss:27778/boot/v1 bss --update-bootparams --image <image>
+./ochami-cli --bss-url http://bss:27778/boot/v1 bss --update-bootparams --kernel <kernel> --initrd <initrd>
+```
+If no `--xname` or `--nid` is specified this will update all the nodes.
+
+You can get the state of BSS with
+```bash
+./ochami-cli --bss-url http://bss:27778/boot/v1 bss --get-bootparams --format {none|json|yaml}
```

Comment:
Might be helpful to say which format is the default and what each mean (e.g. what does `none` mean?).

Reply:
6bc923f